pkg/vault/contracts/authorizer/TimelockAuthorizer.sol

```diff
@@ -635,31 +634,70 @@ contract TimelockAuthorizer is IAuthorizer, IAuthentication, ReentrancyGuard {
     }
 
     /**
-     * @notice Grants or revokes granter status to `account` for action `actionId` in target `where`.
-     * @dev Only the root can add and remove granters.
+     * @notice Grants granter status to `account` for action `actionId` in target `where`.
+     * @dev Only the root can add granters.
      *
-     * Note that there are no delays associated with adding or removing granters. This is based on the assumption that
-     * any action which a malicous user could exploit to damage the protocol will have a sufficiently long delay
-     * associated with either granting permission for or exercising that permission such that the root will be able to
-     * reestablish control and cancel either the granting or associated action before it can be executed, and then
-     * remove the granter.
+     * Note that there are no delays associated with adding granters. This is based on the assumption that any action
+     * which a malicous user could exploit to damage the protocol will have a sufficiently long delay associated with
+     * granting permissions for or exercising that permission such that the root will be able to reestablish control and
+     * cancel either the granting or associated action before it can be executed, and then remove the granter.
      *
      * A malicious granter may also attempt to use their granter status to grant permission to multiple accounts, but
      * they cannot create new granters. Therefore, the danger posed by a malicious granter is limited and self-
      * contained. Root can mitigate the situation simply and completely by revoking first their granter status,
      * and then any permissions granted by that account, knowing there cannot be any more.
      */
-    function manageGranter(
+    function addGranter(
         bytes32 actionId,
         address account,
-        address where,
-        bool allowed
+        address where
+    ) external {
+        require(isRoot(msg.sender), "SENDER_IS_NOT_ROOT");
+
+        require(!isGranter(actionId, account, where), "ACCOUNT_IS_ALREADY_GRANTER");
+        // Note that it is possible for `account` to be a granter for the same `actionId` in some specific `where`, and
+        // then be granted permission over `EVERYWHERE`, resulting in 'duplicate' permissions. This is not an issue per
+        // se, but removing this granter status will require undoing these actions in inverse order.
+        // To avoid these issues, it is recommended to revoke any prior granter status over specific contracts before
+        // making an account a global granter.
```

Comment:
(not strong opinion) maybe consider adding this to natspec as well?
The explanation is a bit long already, but this is also relevant from a UX perspective.